brain/python/synapses.cpp

```diff
@@ -0,0 +1,165 @@
+GET_SYNAPSES_ARRAY_PROPERTY( float, preCenterYPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, preCenterZPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( uint32_t, postGIDs )
+GET_SYNAPSES_ARRAY_PROPERTY( uint32_t, postSectionIDs )
+GET_SYNAPSES_ARRAY_PROPERTY( uint32_t, postSegmentIDs )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postDistances )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postSurfaceXPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postSurfaceYPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postSurfaceZPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postCenterXPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postCenterYPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, postCenterZPositions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, delays )
+GET_SYNAPSES_ARRAY_PROPERTY( float, conductances )
+GET_SYNAPSES_ARRAY_PROPERTY( float, utilizations )
+GET_SYNAPSES_ARRAY_PROPERTY( float, depressions )
+GET_SYNAPSES_ARRAY_PROPERTY( float, facilitations )
+GET_SYNAPSES_ARRAY_PROPERTY( float, decays )
+GET_SYNAPSES_ARRAY_PROPERTY( int, efficacies )
+
+list _toList( const brain::Vector3f& v )
+{
+    list out;
+    out.append( v.x( ));
+    out.append( v.y( ));
+    out.append( v.z( ));
+    return out;
+}
+
+#define GET_SYNAPSE_VEC3_PROPERTY( name ) \
+list Synapse_##name( const SynapseWrapper& synapse ) \
+    { return _toList( synapse.name( )); }
+
+GET_SYNAPSE_VEC3_PROPERTY( getPresynapticSurfacePosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPresynapticCenterPosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPostsynapticSurfacePosition )
+GET_SYNAPSE_VEC3_PROPERTY( getPostsynapticCenterPosition )
+
+}
+
+void export_Synapses()
+{
+
+class_< SynapseWrapper >( "Synapse", no_init )
+    .def( "presynaptic_gid", &Synapse::getPresynapticGID )
+    .def( "presynaptic_section", &Synapse::getPresynapticSectionID )
+    .def( "presynaptic_segment", &Synapse::getPresynapticSegmentID )
+    .def( "presynaptic_distance", &Synapse::getPresynapticDistance )
+    .def( "postsynaptic_gid", &Synapse::getPostsynapticGID )
+    .def( "postsynaptic_section", &Synapse::getPostsynapticSectionID )
+    .def( "postsynaptic_segment", &Synapse::getPostsynapticSegmentID )
+    .def( "postsynaptic_distance", &Synapse::getPostsynapticDistance )
+    .def( "delay", &Synapse::getDelay )
+    .def( "conductance", &Synapse::getConductance )
+    .def( "utilization", &Synapse::getUtilization )
+    .def( "depression", &Synapse::getDepression )
+    .def( "facilitation", &Synapse::getFacilitation )
+    .def( "decay", &Synapse::getDecay )
+    .def( "efficacy", &Synapse::getEfficacy )
+    ;
+
+class_< SynapsesWrapper >( "Synapses", no_init )
+    .def( "__nonzero__", nonzero )
+    .def( "__len__", &Synapses::size )
+    .def( "__getitem__", Synapses_get )
+    // There is no need to wrap the iterator, Python provides one out of the
+    // box thanks to __len__ and __getitem__
+    .def( "empty", &Synapses::empty )
+    .def( "indices", Synapses_indices )
+    .def( "pre_gids", Synapses_preGIDs )
```

Comment:
where? All the Synapse methods use _ .
What I've realized is that there were 4 methods missing in the wrapping.